internal/validate/pod.go

```diff
@@ -0,0 +1,90 @@
+package validate
+
+import (
+	"context"
+	"github.com/kyma-project/warden/internal/util/sets"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/warden/pkg"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ValidationResult int
+
+const (
+	Invalid ValidationResult = iota
+	ServiceUnAvailable
+	Valid
+	NoAction
+)
+
+type PodValidator interface {
+	ValidatePod(ctx context.Context, pod *corev1.Pod, ns *corev1.Namespace) (ValidationResult, error)
+}
+
+type NamespaceChecker interface {
+	IsValidationEnabledForNS(namespace string) bool
+}
+
+var _ PodValidator = &podValidator{}
+
+type podValidator struct {
+	Validator ImageValidatorService
+}
+
+func NewPodValidator(imageValidator ImageValidatorService) PodValidator {
+	return &podValidator{
+		imageValidator,
+	}
+}
+
+func (a *podValidator) ValidatePod(ctx context.Context, pod *corev1.Pod, ns *corev1.Namespace) (ValidationResult, error) {
+	l := log.FromContext(ctx)
```

Comment:
If there is no plan for `ctx` we could remove it.

Reply:
There is a plan for `ctx`, so I will keep it.